nexus/src/app/sagas/volume_delete.rs

```diff
@@ -422,7 +433,25 @@ async fn svd_delete_freed_crucible_regions(
             },
         )?;
 
-    for (dataset, region, region_snapshot, volume) in
+    // Don't serialize the whole Volume, as the data field contains key material!
+    Ok(freed_datasets_regions_and_volumes
+        .into_iter()
+        .map(|x| (x.0, x.1, x.2, x.3.id()))
+        .collect())
+}
+
+async fn svd_delete_freed_crucible_regions(
+    sagactx: NexusActionContext,
+) -> Result<(), ActionError> {
+    let log = sagactx.user_data().log();
+    let osagactx = sagactx.user_data();
+
+    // Find regions freed up for deletion by a previous saga node deleting the
+    // region snapshots.
+    let freed_datasets_regions_and_volumes =
+        sagactx.lookup::<FreedCrucibleRegions>("freed_crucible_regions")?;
```

Comment:
this has the nice secondary benefit of giving us a record of which regions the saga will try to delete, right? i'd had a note to make sure we log that which i now can strike..

Reply:
You are correct :)